core/com.dexels.navajo.function/src/com/dexels/navajo/functions/Min.java

```diff
@@ -66,17 +64,24 @@ public final Object evaluate() throws TMLExpressionException {
         } else
             throw new TMLExpressionException("Invalid number of arguments for Max()");
     }
-    private Object min(Object a, Object b)
+
+    private Object min( Object a, Object b )
     {
-        if (a instanceof Date)
+        if ( a != null && ( a instanceof Date || a instanceof ClockTime ) )
```

Comment:
you can leave out the null check, see: https://stackoverflow.com/questions/2950319/is-null-check-needed-before-calling-instanceof

Reply:
The null check is not there for the instanceof. Right now if you enter a Date and a null value you end up at the coded exception "Cannot compare a date with a non-date". But it should be possible to compare these with the Date always winning, both at Min and Max.